deploy/kubernetes/schema.go

```diff
@@ -0,0 +1,1783 @@
+	),
+	// endregion
+	// region AzureDiskVolumeSource
+	schema.NewStructMappedObjectSchema[v1.AzureDiskVolumeSource](
+		"AzureDiskVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region PhotonPersistentDiskVolumeSource
+	schema.NewStructMappedObjectSchema[v1.PhotonPersistentDiskVolumeSource](
+		"PhotonPersistentDiskVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region ProjectedVolumeSource
+	schema.NewStructMappedObjectSchema[v1.ProjectedVolumeSource](
+		"ProjectedVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region PortworxVolumeSource
+	schema.NewStructMappedObjectSchema[v1.PortworxVolumeSource](
+		"PortworxVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region ScaleIOVolumeSource
+	schema.NewStructMappedObjectSchema[v1.ScaleIOVolumeSource](
+		"ScaleIOVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region StorageOSVolumeSource
+	schema.NewStructMappedObjectSchema[v1.StorageOSVolumeSource](
+		"StorageOSVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region CSIVolumeSource
+	schema.NewStructMappedObjectSchema[v1.CSIVolumeSource](
+		"CSIVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+	// region EphemeralVolumeSource
+	schema.NewStructMappedObjectSchema[v1.EphemeralVolumeSource](
+		"EphemeralVolumeSource",
+		map[string]*schema.PropertySchema{},
+	),
+	// endregion
+)
+
+func generateVolumeTypeList(except string) []string {
+	l := []string{
+		"hostPath",
+		"emptyDir",
+		"gcePersistentDisk",
+		"awsElasticBlockStore",
+		"secret",
+		"nfs",
+		"iscsi",
+		"glusterfs",
+		"persistentVolumeClaim",
+		"rbd",
+		"flexVolume",
+		"cinder",
+		"cephfs",
+		"flocker",
+		"downwardAPI",
+		"fc",
+		"azureFile",
+		"configMap",
+		"vsphereVolume",
+		"quobyte",
+		"azureDisk",
+		"photonPersistentDisk",
+		"projected",
+		"portworxVolume",
+		"scaleIO",
+		"storageos",
+		"csi",
+		"ephemeral",
+	}
+	var result []string
+	for _, entry := range l {
+		if entry != except {
+			result = append(result, entry)
+		}
+	}
+	return result
+}
+
+var identifier = schema.NewStringSchema(
+	schema.IntPointer(1),
+	nil,
+	regexp.MustCompile(`^[a-zA-Z0-9-._]+$`),
+)
+var imageTag = schema.NewStringSchema(
+	schema.IntPointer(1),
+	nil,
+	regexp.MustCompile(`^[a-zA-Z0-9_\-:.]+$`),
+)
+var labelName = schema.NewStringSchema(
+	nil,
+	nil,
+	regexp.MustCompile(`(|([a-zA-Z](|[a-zA-Z\-.]{0,251}[a-zA-Z0-9]))/)([a-zA-Z](|[a-zA-Z\\-]{0,61}[a-zA-Z0-9]))`),
+)
+var labelValue = schema.NewStringSchema(
+	nil,
+	schema.IntPointer(63),
+	regexp.MustCompile(`^(|[a-zA-Z0-9]+(|[-_.][a-zA-Z0-9]+)*[a-zA-Z0-9])$`),
+)
+var dnsSubdomainName = schema.NewStringSchema(
+	nil,
+	schema.IntPointer(253),
+	regexp.MustCompile(`^[a-z0-9]($|[a-z0-9\-_]*[a-z0-9])$`),
+)
+
+//nolint:deadcode,unused,varcheck
+var rfc1123LabelName = schema.NewStringSchema(
```

Comment:
This is going to be used in an early follow-up PR, right? If it is not sure, I'd rather remove it here and add it when needed (YAGNI) - otherwise this is dead code and won't be detected due to the `nolint`.